app/javascript/src/Stats/inlinechart/index.jsx

```diff
@@ -0,0 +1,145 @@
+// @flow
+
+import React, {Component } from 'react'
+import type { Node } from 'react'
+import c3 from 'c3'
+import 'whatwg-fetch'
+import 'core-js/es6/promise'
+import 'url-polyfill'
+import moment from 'moment'
+import numeral from 'numeral'
+
+type Props = {
+    endPoint: string,
+    metricName: string,
+    title: string,
+    unitPluralized: string
+}
+
+type State = {
+    loading: boolean,
+    title: string,
+    total: string,
+    values: Array<number>,
+    unit: string,
+    unitPluralized: string
+}
+
+type RefObject = { current: null | HTMLDivElement }
+
+type DataMetric = {
+    unit: string
+}
+
+type Data = {
+    metric: DataMetric,
+    total: number,
+    values: Array<number>
+}
+
+type ResponseError = {
+  status: string,
+  statusText: string
+}
+
+class InlineChart extends Component<Props, State> {
+  c3ChartContainer: RefObject
+
+  constructor (props: Props) {
+    super(props)
+    this.c3ChartContainer = React.createRef()
+    this.state = {
+      loading: true,
+      title: this.props.title,
+      total: '',
+      values: [],
+      unit: '',
+      unitPluralized: this.props.unitPluralized
+    }
+  }
+
+  generateC3Chart () {
+    const nodeElem = this.c3ChartContainer.current
+    const optionsObj = {
+      bindto: nodeElem,
+      axis: {
+        x: { show: false },
+        y: { show: false }
+      },
+      legend: { show: false },
+      point: { show: false },
+      data: {
+        columns: [
+          [...this.state.values]
+        ]
+      },
+      tooltip: {
+        contents: function (d) {
+          return `<span><i class='tooltip-dot'></i> ${d[0].value}</span>`
+        }
+      },
+      onresize: function () {
+        nodeElem.style.maxHeight = 'none'
+      }
+    }
+    c3.generate(optionsObj)
+  }
+
+  getTotalAsString (total: number, unit: string) {
+    return `${numeral(total).format('0,0')} ${unit.substring(0, 10)}`
+  }
+
+  updateState (data: Data) {
+    let unit = data.metric.unit
+    if (unit.slice(-1) !== 's' && data.total !== 1) {
+      unit = this.state.unitPluralized
+    }
+    const total = this.getTotalAsString(data.total, unit)
+
+    this.setState({
+      loading: false,
+      total,
+      unit,
+      values: data.values
+    }, () => this.generateC3Chart())
+  }
+
+  getURL (): URL {
```

Comment:
you should test this URL crafting, domain, base and params

Reply:
done ;)